vip-jetpack/vip-jetpack.php

```diff
@@ -409,3 +409,36 @@ function vip_filter_plugin_version_jetpack( $plugin_meta, $plugin_file ) {
 	return $plugin_meta;
 }
 add_filter( 'plugin_row_meta', 'vip_filter_plugin_version_jetpack', PHP_INT_MAX, 2 );
+
+/**
+ * Enable Jetpack offline mode for Multisites when we're launching a site in the network.
+ * This is enabled only if the site is a multisite and they enrolled into the JETPACK_SYNC_IDC_OPTIN.
+ * 
+ * The function checks for the `launching` flag in the `vip_launch_tools` cache group.
+ * If the flag is set to `true`, it enables the offline mode while the flag is active, blocking 
+ * the Jetpack communications from running.
+ * @param bool $offline_mode Whether to enable offline mode.
+ * @return bool
+ */
+function vip_filter_jetpack_offline_mode_on_site_launch( $offline_mode ) {
+	// If not multisite, return the offline mode value.
+	if (! is_multisite() ){
+		return $offline_mode;
+	}
+
+	// We want to change the offline mode only if the customer enrolls into the IDC resolution
+	if ( ! defined( 'JETPACK_SYNC_IDC_OPTIN' ) || true !== constant( 'JETPACK_SYNC_IDC_OPTIN' ) ) {
+		return $offline_mode;
+	}
+	if ( wp_cache_get( 'launching', 'vip_launch_tools' ) ){
```

Comment:
@WPprodigy I'm using the `wp_cache_get` here as you suggested. I wonder, though, aside from interacting with the WP CLI cache commands, could this option be cleared out by some other tools we have? If so, which ones?

I want to be sure we don't enter in online mode by accident when we're launching a site.